### PR TITLE
Python make generated struct hashable

### PR DESCRIPTION
**Pull request by brianshannan-wf**
Lists, sets, and maps aren't hashable in python. This converts said structures to a hashable form so the `__hash__` function should work in the generated code.

@Workiva/infre-product @Workiva/messaging-pp @sagesmith-wf @theisensanders-wf

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Raven
<details>
<summary>Number of Findings: 0</summary>

</details>

## Magpie
<details>
<summary>Number of findings: 0</summary>

</details>

**Comment by stevenosborne-wf** [comment]
+1

File: lib/python/frugal/util/__init__.py [hunk 1]
```diff
@@ -8,3 +8,24 @@
...
+    :param thing: The data to create a hashable representation of.
+    """
+    if isinstance(thing, list):
+        return tuple([make_hashable(elem) for elem in thing])
+    elif isinstance(thing, set):
+        return frozenset([make_hashable(elem) for elem in thing])
+    elif isinstance(thing, dict):
+        new_dict = {make_hashable(k): make_hashable(v)
+                    for k, v in six.iteritems(thing)}
+        return frozenset(six.iteritems(new_dict))
```
Comment (garrickpeterson-wf) [inline lib/python/frugal/util/__init__.py]: This creates a dictionary in memory and immediately throws it away. Can we instead just put the generator right into the frozenset constructor?

```python
return frozenset((make_hashable(k), make_hashable(v)) for k, v in six.iteritems(thing))
```

File: lib/python/frugal/util/__init__.py [hunk 2]
```diff
@@ -8,3 +8,24 @@
...
+
+def make_hashable(thing):
+    """
+    Creates a representation of some input data with immutable collections,
+    so the output is hashable.
+
+    :param thing: The data to create a hashable representation of.
+    """
+    if isinstance(thing, list):
+        return tuple([make_hashable(elem) for elem in thing])
```
Comment (garrickpeterson-wf) [inline lib/python/frugal/util/__init__.py]: The list comprehension is not required, you can simply do `tuple(make_hashable(elem) for elem in thing)`

Ditto for the frozenset below.

**Comment by stevenosborne-wf** [comment]
+1

**Comment by garrickpeterson-wf** [comment]
+1

**Pull request merged**